abce/agent.py

```diff
@@ -48,6 +48,112 @@
 import traceback
 from abce.notenoughgoods import NotEnoughGoods
 import datetime
+from abce.trade import get_epsilon
+epsilon = get_epsilon()
+
+
+class Inventory(defaultdict):
+    def __init__(self, name):
+        super(Inventory, self).__init__(float)
+        self.name = name
+        self.contracts = set()
+        self._expiring_goods = []
+        self._perishable = []
+        #self.goods = defaultdict(float)
```

Comment:
Yes that is the direction. Currently, it defaults to `defaultdict` so that the rest of the code still works.